crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1829,6 +1829,10 @@ impl AuthorityPerEpochStore {
         ))
     }
 
+    // Caller is not required to set ExecutionIndices with the right semantics in
+    // VerifiedSequencedConsensusTransaction.
+    // Also, ConsensusStats and hash will not be updated in the db with this function, unlike in
```

Comment:
hmm I made an attempt to merge them in https://github.com/MystenLabs/sui/pull/14126, would that not work?

Reply:
I think that should be fine. I don't think any test is checking the consensus indices and stats written after a consensus commit.

Reply:
I will leave that refactor to your PR if it's ok :)